### PR TITLE
Add Debian 10 support

### PR DESCRIPTION
**Pull request by martijndegouw**

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -217,4 +217,9 @@
     'TrackSlurmctldDown' => 'no',
     'TrackWCKey' => undef,
   }
+
+  $env_dir = $::$::osfamily ? {
```
Comment (treydock) [inline manifests/params.pp:221]: per comments in #22 this should be in `init.pp` as a parameter and then change the default value using Hiera in `data/os`.

File: manifests/slurmctld/service.pp [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # @api private
 class slurm::slurmctld::service {
 
-  file { '/etc/sysconfig/slurmctld':
+  file { "${slurm::params::env_dir}/slurmctld":
```
Comment (treydock) [inline manifests/slurmctld/service.pp:4]: Once this is in `init.pp` this would just be `${slurm::env_dir}`

File: manifests/common/install/apt.pp [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    #      gpgcheck        => '0',
+    #      priority        => '1',
+    #    }
+  } else {
+    $package_require = undef
+  }
+
+  Package {
+    ensure  => $slurm::package_ensure,
+    require => $package_require,
```
Comment (treydock) [inline manifests/common/install/apt.pp:21]: Can remove this `require` definition once above condition is removed.

**Comment by martijndegouw** [comment]
Thanks a lot for your feedback. I updated the PR already with your initial comments, but it needs more testing as well from my side.

**Comment by martijndegouw** [comment]
This version seems to work at my place :)

File: manifests/init.pp [hunk 4]
```diff
@@ -261,6 +262,7 @@
   # Managed directories
   Stdlib::Absolutepath $conf_dir = '/etc/slurm',
   Stdlib::Absolutepath $log_dir  = '/var/log/slurm',
+  Stdlib::Absolutepath $env_dir,
```
Comment (treydock) [inline manifests/init.pp:265]:
```suggestion
  Stdlib::Absolutepath $env_dir = '/etc/sysconfig',
```
Make the default have a value

**Comment by treydock** [comment]
This and #23 will have their acceptance tests fail because the default SLURM version is this module is no longer available due to some CVEs.  So I am bumping the version in #25 and hopefully that will fix the acceptance tests.

**Comment by treydock** [comment]
Go ahead and merge master branch or rebase against master to pick up fix for acceptance tests.

**Pull request merged**